src/main/java/org/jitsi/srtp/SrtcpCryptoContext.java

```diff
@@ -202,9 +186,96 @@ private void processPacketAesCm(ByteArrayBuffer pkt, int index)
+         *       +--+--+--+--+--+--+--+--+--+--+--+--+
+         *       |00|00|    SSRC   |00|00|0+SRTCP Idx|---+
+         *       +--+--+--+--+--+--+--+--+--+--+--+--+   |
+         *                                               |
+         *       +--+--+--+--+--+--+--+--+--+--+--+--+   |
+         *       |         Encryption Salt           |->(+)
+         *       +--+--+--+--+--+--+--+--+--+--+--+--+   |
+         *                                               |
+         *       +--+--+--+--+--+--+--+--+--+--+--+--+   |
+         *       |       Initialization Vector       |<--+
+         *       +--+--+--+--+--+--+--+--+--+--+--+--+
+         */
+
+        ivStore[0] = saltKey[0];
+        ivStore[1] = saltKey[1];
+
+        // The shifts transform the ssrc and index into network order
+        ivStore[2] = (byte) (((ssrc >> 24) & 0xff) ^ saltKey[2]);
+        ivStore[3] = (byte) (((ssrc >> 16) & 0xff) ^ saltKey[3]);
+        ivStore[4] = (byte) (((ssrc >> 8) & 0xff) ^ saltKey[4]);
+        ivStore[5] = (byte) ((ssrc & 0xff) ^ saltKey[5]);
+
+        ivStore[6] = saltKey[6];
+        ivStore[7] = saltKey[7];
+
+        ivStore[8] = (byte) (((index >> 24) & 0xff) ^ saltKey[8]);
+        ivStore[9] = (byte) (((index >> 16) & 0xff) ^ saltKey[9]);
+        ivStore[10] = (byte) (((index >> 8) & 0xff) ^ saltKey[10]);
+        ivStore[11] = (byte) ((index & 0xff) ^ saltKey[11]);
+
+        try
+        {
+            cipher.setIV(ivStore, forEncryption);
+
+            if (encrypt)
+            {
+                // Need the encryption flag
+                index = index | 0x80000000;
+
+                // Encrypted part excludes fixed header (8 bytes)
+                int payloadOffset = 8;
+
+                cipher.processAAD(pkt.getBuffer(), pkt.getOffset(),
+                    payloadOffset);
+                writeRoc(index);
+                cipher.processAAD(rbStore, 0, 4);
+
+                int processLen = cipher.process(
+                    pkt.getBuffer(),
+                    pkt.getOffset() + payloadOffset,
+                    pkt.getLength() - payloadOffset);
+
+                pkt.setLength(processLen + payloadOffset);
+            }
+            else
+            {
+                int bufferedTagLen = forEncryption ? 0 : policy.getAuthTagLength();
+                int aadLen = pkt.getLength() - bufferedTagLen;
+                if (aadLen < 0)
+                {
+                    return SrtpErrorStatus.AUTH_FAIL;
+                }
+                cipher.processAAD(pkt.getBuffer(), pkt.getOffset(), aadLen);
+                writeRoc(index);
+                cipher.processAAD(rbStore, 0, 4);
+
+                int processLen = cipher.process(
+                    pkt.getBuffer(), aadLen, bufferedTagLen);
+
+                pkt.setLength(aadLen + processLen);
+            }
+        }
+        catch (GeneralSecurityException e)
+        {
+            return SrtpErrorStatus.AUTH_FAIL;
```

Comment:
if `forEncyption==true` => `SrtpErrorStatus.FAIL`? And should the exception be (debug-level) logged?